packages/babel-plugin-transform-no-undefined-identifier/src/index.js

```diff
@@ -0,0 +1,27 @@
+export default function() {
+  return {
+    visitor: {
+      CallExpression(path, state) {
+        const args = path.get("arguments");
+        const evaluatedCallee = path.get("callee").evaluate();
+
+        if (!state.file.opts.filename.includes("packages/babel-plugin-")) {
+          return;
+        }
+
+        if (
+          args.length &&
+          args[0].evaluate().value === "undefined" &&
+          evaluatedCallee.deopt.node.object &&
+          evaluatedCallee.deopt.node.object.name === "t" &&
```

Comment:
is this the best way to evaluate a reassigned `MemberExpression`? tried to reverse-engineer this from other sourcecode as best i could -- if there's a better way to do this please let me know!

Reply:
It still doesn't catch every case (like `const a = t; a.identifier`) but it is smart enought to disallow `const buildIdentifier = t.identifier; buildIdentifier("undefined")`, which is the only sensible case I can think of where the call is not `t.identifier()`

Reply:
yeah. i was thinking that bit might not be the most necessary. regardless, i'm gonna play with it a bit more and try to cover that too, if for no reason other than to just learn more about writing babel plugins 😝